src/apps/explorer/components/TransactionsTableWidget/index.tsx

```diff
@@ -24,30 +25,52 @@ interface Props {
   transactions?: Order[]
 }
 
-const tabItems = (isLoadingOrders: boolean): TabItemInterface[] => {
+enum TabViews {
+  GRAPH = 'graph',
+  ORDERS = 'orders',
+}
+
+const TAB_VIEW_ID = {
+  [TabViews.ORDERS]: 1,
+  [TabViews.GRAPH]: 2,
+}
+
+const DEFAULT_TAB = TabViews.ORDERS
+
+function useQueryViewParams(): { tab: string } {
+  const query = useQuery()
+  return { tab: query.get('tab') || DEFAULT_TAB }
+}
+
+const tabItems = (txBatchTrades: GetTxBatchTradesResult, networkId: BlockchainNetwork): TabItemInterface[] => {
   return [
     {
-      id: 1,
-      tab: (
-        <>
-          Orders
-          <StyledTabLoader>{isLoadingOrders && <Spinner spin size="1x" />}</StyledTabLoader>
-        </>
-      ),
+      id: TAB_VIEW_ID[TabViews.ORDERS],
+      tab: <TabIcon title="Orders" iconFontName={faListUl} />,
       content: <TransactionsTableWithData />,
     },
+    {
+      id: TAB_VIEW_ID[TabViews.GRAPH],
+      tab: <TabIcon title="Graph" iconFontName={faProjectDiagram} />,
+      content: <TransactionBatchGraph txBatchData={txBatchTrades} networkId={networkId} />,
+    },
   ]
 }
 
 export const TransactionsTableWidget: React.FC<Props> = ({ txHash }) => {
   const { orders, isLoading: isTxLoading, errorTxPresentInNetworkId, error } = useGetTxOrders(txHash)
   const networkId = useNetworkId() || undefined
   const [redirectTo, setRedirectTo] = useState(false)
-  const [batchViewer, setBatchViewer] = useState(false)
+  const { tab } = useQueryViewParams()
+  const [tabViewName, setTabViewName] = useState<TabViews>(
+    (Object.values(TabViews).includes(tab as TabViews) && (tab as TabViews)) || TabViews.ORDERS,
+  )
```

Comment:
It's an enum, you can access it via `TabViews[tab]`

```suggestion
  const [tabViewName, setTabViewName] = useState<TabViews>(TabViews[tab] || TabViews.ORDERS)
```